meshio/msh_io/main.py

```diff
@@ -22,22 +22,41 @@ def read_buffer(f):
     int_size = sizeof(c_int)
     format_version, data_size, is_ascii = _read_header(f, int_size)
 
-    assert format_version in ["2", "4"], "Need mesh format 2 or 4 (got {})".format(
-        format_version
-    )
+    readers = {2: msh2, 4.0: msh4_0, 4.1: msh4_1}
```

Comment:
I guess `2`, `4.0`, `4.1` should be strings. The implicit float comparison in `readers[format_version]` makes me a bit nervous.

Reply:
Yes.  506685abbf715598b714160030a19231e0eaca52

also sharing this `Dict[str, Module]` with `msh_io.write`.

I note that according to the spec, in [MSH file format version 2](http://gmsh.info/dev/doc/texinfo/gmsh.html#MSH-file-format-version-2)
> `version-number` is a real number equal to 2.2

from which I infer that 2.0 and 2.1 are unsupported, but I'll leave that aside for the moment, treating all 2.x the same.